thrust/testing/transform_input_output_iterator.cu

```diff
@@ -7,8 +7,16 @@
 
 #include <unittest/unittest.h>
 
+// There is a unfortunate miscompilation of the gcc-11 vectorizer leading to OOB writes
+// Adding this attribute suffices that this miscompilation does not appear anymore
+#if defined(_CCCL_COMPILER_GCC) && __GNUC__ >= 11
+#  define THRUST_DISABLE_BROKEN_GCC_VECTORIZER __attribute__((optimize("no-tree-vectorize")))
+#else
+#  define THRUST_DISABLE_BROKEN_GCC_VECTORIZER
+#endif
+
 template <class Vector>
-void TestTransformInputOutputIterator()
+THRUST_DISABLE_BROKEN_GCC_VECTORIZER void TestTransformInputOutputIterator()
```

Comment:
This fixes our tests, but won't gcc still be miscompiling Thrust for users?

Reply:
That is nothing we can change. I want to note that this is exceptionally frickle and dependent on exact sizes and otimization settings, so I dont see anything we can do there